lib/chef/provider/aws_dhcp_options.rb

```diff
@@ -15,7 +15,7 @@ def create_aws_object
 
     converge_by "create DHCP options #{new_resource.name} in #{region}" do
       dhcp_options = new_resource.driver.ec2.dhcp_options.create(options)
-      retry_with_backoff(AWS::EC2::Errors::InvalidDhcpOptionsID::NotFound) do
+      retry_with_backoff(Aws::EC2::Errors::InvalidDhcpOptionsID::NotFound) do
```

Comment:
A recommendation: Whenever switching from `AWS` to `Aws` we should always prepend it with `::` - so this would be `::Aws::EC2::Errors::InvalidDhcpOptionsID::NotFound`.

AWS generates their classes on the fly (last I checked) so there is no file we can require. Without requiring the file that exposes these classes, we need to make sure Ruby knows we are referencing the base `Aws` module and not a sub-module of the current location.